docs/scripts/link-checker-puppeteer.ts

```diff
@@ -0,0 +1,72 @@
+import { PromisePool } from '@supercharge/promise-pool';
+import { sitePaths } from '../src/data/sitePaths';
+import { checkLink, crawlAllLinks } from './util';
+import type { LinkInfo } from './util';
+
+/**
+ * Divide the sitePaths array so that we can easily run a smaller portion if needed.
+ * e.g. to run the first 10 links. run `yarn docs node --require esbuild-register ./scripts/link-checker-puppeteer.ts 0 10`
+ */
+const start = process.argv[2] ?? 0;
+const end = process.argv[3];
+const testPaths = end ? sitePaths.slice(+start, +end) : sitePaths.slice(+start);
+
+try {
+  runLinkChecker();
+} catch (err) {
+  process.exit(1);
+}
+
+async function runLinkChecker() {
+  const allPagesPaths = await crawlAllLinks(testPaths);
+
+  const { results } = await PromisePool.withConcurrency(10)
```

Comment:
Continue the conversation https://github.com/aws-amplify/amplify-ui/pull/3494#discussion_r1123587133

Thanks for suggesting [PromisePool](https://www.npmjs.com/package/@supercharge/promise-pool), @ioanabrooks ! The time of the test is significantly reduced from about 480s on Github to 170s. And only takes 60s on my cloud desktop 🎉

Code is updated.